Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -3199,6 +3256,13 @@ def visit_TryFinallyStatNode(self, node):
             node.is_terminator = True
         return node
 
+    def visit_PassStatNode(self, node):
+        """Eliminate useless PassStatNode"""
+        # 'pass' statements often appear in a separate line and must be traced.
+        if not self.current_directives['linetrace']:
+            node = Nodes.StatListNode(pos=node.pos, stats=[])
```

Comment:
I previously had something different, which didn't work well:
https://github.com/cython/cython/pull/6144/commits/e49c5b8aef9933c28a1b29003dae5f6e5bb68a5b

It's a bit more subtle: we usually need *some* node where we had pass statements (there are probably cases where a simple `None` would bother us), but I wanted to avoid nested empty `StatListNodes` to avoid bloating the overall tree structure. Thus the "put in an empty `StatListNode` but not inside another `StatListNode`" because we definitely don't need it there.

I now took the opportunity to generalise this so that we flatten all `StatListNodes`, rather than just discarding nested empty ones. Let's see if that works.

Reply:
It doesn't work to flatten the `StatListNode` lists, not just like that. I've reverted the change.